pkg/api/validation/dynakube/databases.go

```diff
@@ -0,0 +1,108 @@
+	errorInvalidDatabasesVolumeMount      = `Invalid database volume mount detected: %s. No matching database volume found.`
+	errorUnusedDatabasesVolumes           = `Unused database volume(s) found (%s). Make sure to mount all database volumes defined in the DynaKube.`
+	errorInvalidDatabasesVolumeMountPath  = `Invalid database volume mount path detected (%s). Make sure to use absolute paths.`
+
+	warningHostPathDatabaseVolumeDetected = `Host path database volume detected. If you're on OpenShift, mounting host path volumes will be prohibited by the SCC and cause silent failures. If you still want to do this, make sure to create and bind corresponding roles.`
+)
+
+func conflictingOrInvalidDatabasesVolumeMounts(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	if !dk.Extensions().IsDatabasesEnabled() {
+		return ""
+	}
+
+	defaultVolumeMounts := databases.GetDefaultVolumeMounts(dk)
+	for _, database := range dk.Spec.Extensions.Databases {
+		for _, volumeMount := range database.VolumeMounts {
+			volumeFound := slices.IndexFunc(database.Volumes, func(volume corev1.Volume) bool {
+				return volume.Name == volumeMount.Name
+			}) >= 0
+			if !volumeFound {
+				log.Info("invalid database volume mount detected; no matching volume found", "volume", volumeMount.Name)
+
+				return fmt.Sprintf(errorInvalidDatabasesVolumeMount, volumeMount.Name)
+			}
+
+			for _, defaultVolumeMount := range defaultVolumeMounts {
+				rel, err := filepath.Rel(defaultVolumeMount.MountPath, volumeMount.MountPath)
+				if err != nil {
+					return fmt.Sprintf(errorInvalidDatabasesVolumeMountPath, volumeMount.MountPath)
+				}
+
+				if !strings.HasPrefix(rel, "../") {
+					log.Info("conflicting database volume mount detected; clashes with pre-defined volume", "path", volumeMount.MountPath)
+
+					return fmt.Sprintf(errorConflictingDatabasesVolumeMounts, volumeMount.MountPath, defaultVolumeMount.MountPath)
+				}
+			}
+		}
+	}
+
+	return ""
+}
+
+func hostPathDatabaseVolumeFound(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	if !dk.Extensions().IsDatabasesEnabled() {
+		return ""
+	}
+
+	for _, database := range dk.Spec.Extensions.Databases {
+		for _, volume := range database.Volumes {
+			if volume.HostPath != nil {
+				log.Info("host path volume detected, this may cause issues on openshift", "volumeName", volume.Name)
+
+				return warningHostPathDatabaseVolumeDetected
+			}
+		}
+	}
+
+	return ""
+}
+
+func unusedDatabasesVolume(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	if !dk.Extensions().IsDatabasesEnabled() {
+		return ""
+	}
+
+	unusedVolumes := make([]string, 0)
+
+	for _, database := range dk.Spec.Extensions.Databases {
+		for _, volume := range database.Volumes {
+			volumeUsed := false
+
+			for _, volumeMount := range database.VolumeMounts {
```

Comment:
this can also be expressed as slices.Contains(Func)

Reply:
Thanks, addressed in 4ca52f2e9 😄